src/python/tools/preprocess/converters/torch_converter.py

```diff
@@ -412,11 +480,23 @@ def __init__(
         self.output_dir = output_dir
         self.num_nodes = num_nodes
         self.num_rels = num_rels
+        self.columns = columns
+
+        # Determine the edge type column id and the weight type column id
+        self.edge_type_column = -1
+        self.edge_weight_column = edge_weight_column
+        if len(self.columns) > 2:
+            self.edge_type_column = self.columns[1]
+        print("Edge type column", self.edge_type_column, "with edge weight column", self.edge_weight_column)
 
         if format.upper() in SUPPORTED_DELIM_FORMATS:
             assert isinstance(train_edges, str) or isinstance(train_edges, Path)
 
-            self.reader = PandasDelimitedFileReader(train_edges, valid_edges, test_edges, columns, header_length, delim)
+            max_col_num = max(max(self.columns), self.edge_weight_column)
+            all_cols_to_read = [i for i in range(0, max_col_num + 1)]
```

Comment:
confused by this logic. 

For example say the edge_weight_column is column 9 while the src and dst columns are in column 0 and column 1. This logic suggests we will read columns 0-9 rather than just 0, 1 and 9.